pkg/operator/ceph/cluster/mon/config.go

```diff
@@ -184,19 +184,19 @@ func createClusterAccessSecret(clientset kubernetes.Interface, namespace string,
 	var err error
 
 	// store the secrets for internal usage of the rook pods
-	secrets := map[string]string{
-		clusterSecretName: clusterInfo.Name,
-		fsidSecretName:    clusterInfo.FSID,
-		monSecretName:     clusterInfo.MonitorSecret,
-		adminSecretName:   clusterInfo.AdminSecret,
+	secrets := map[string][]byte{
+		clusterSecretName: []byte(clusterInfo.Name),
+		fsidSecretName:    []byte(clusterInfo.FSID),
+		monSecretName:     []byte(clusterInfo.MonitorSecret),
+		adminSecretName:   []byte(clusterInfo.AdminSecret),
 	}
 	secret := &v1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      appName,
 			Namespace: namespace,
 		},
-		StringData: secrets,
-		Type:       k8sutil.RookType,
+		Data: secrets,
```

Comment:
Haha! This was a funny issue for me to debug. The way the kubernetes mock works for unit tests, writing to `StringData` doesn't put the data into `Data` like the real kubernetes. It's necessary to read From `Data` since `StringData` is input-only in k8s, and by extension, it's necessary to write to `Data` for unit tests.